src/main/java/org/terasology/moduletestingenvironment/TestEventReceiver.java

```diff
@@ -63,64 +74,79 @@
-     *
+     * <p>
+     * The following signature of a {@code TestEventReceiver} is equivalent to the event handler below:
      * <pre>
-     * {@literal
-     * TestEventReceiver receiver = new TestEventReceiver<>(context, DropItemEvent.class, (event, entity) -> {
+     * {@code
+     * TestEventReceiver receiver = new TestEventReceiver<>(context, MyEvent.class, (event, entity) -> {
      *   // do something with the event or entity
-     * });
+     * }, MyComponent.class);
+     *
+     * // ... corresponds to
+     *
+     * @ReceiveEvent(components = {MyComponent.class})
+     * public void handler(MyEvent event, EntityRef entity) {
+     *     // do something with the event or entity
+     * }
      * }
      * </pre>
      *
-     * @param context the context object for the test; this should probably be obtained through
-     *                {@link ModuleTestingEnvironment#getHostContext()} and is needed so we can
-     *                obtain an {@link EventSystem} instance to register our event handler.
-     * @param eventClass   the {@link Event} subclass to listen for
-     * @param handler an optional {@link BiConsumer} to fire when events are received.
+     * @param context the context object for the test; this should probably be obtained through {@link
+     *         ModuleTestingHelper#createClient()} and is needed so we can obtain an {@link EventSystem} instance to
+     *         register our event handler.
+     * @param eventClass the {@link Event} subclass to listen for
+     * @param handler an optional {@link BiConsumer} fired when events are received
+     * @param componentTypes list of component types that need to be present on the entity receiving the event
      */
-    public TestEventReceiver(Context context, Class<T> eventClass, BiConsumer<T, EntityRef> handler) {
-        this.handler = handler;
+    public TestEventReceiver(Context context, Class<T> eventClass, BiConsumer<T, EntityRef> handler, Class<?
+            extends Component>... componentTypes) {
         this.eventClass = eventClass;
+        this.handler = handler;
         eventSystem = context.get(EventSystem.class);
-        eventSystem.registerEventReceiver(this, eventClass, EntityInfoComponent.class);
+
+        Class<? extends Component>[] components =
+                Lists.asList(EntityInfoComponent.class, componentTypes).toArray(new Class[componentTypes.length + 1]);
+
+        eventSystem.registerEventReceiver(this, eventClass, components);
     }
 
     /**
-     * @see #TestEventReceiver(Context, Class, BiConsumer)
+     * @see #TestEventReceiver(Context, Class, BiConsumer, Class[])
      */
     public TestEventReceiver(Context context, Class<T> eventClass) {
-        this.eventClass = eventClass;
-        eventSystem = context.get(EventSystem.class);
-        eventSystem.registerEventReceiver(this, eventClass, EntityInfoComponent.class);
+        this(context, eventClass, (event, entity) -> {
```

Comment:
I removed the duplicate code for initializing the event receiver by calling the more general constructor instead.